src/main/java/pl/allegro/tech/discovery/consul/recipes/watch/Disposable.java

```diff
@@ -0,0 +1,13 @@
+package pl.allegro.tech.discovery.consul.recipes.watch;
+
+public class Disposable {
```

Comment:
`Disposable` sounds like an interface that a class can implement if it supports cleaning itself. This is not what we're doing here. We have a class for objects that allow cancelling some processes. You even have a method `isCancelled`, not `isDisposed`. I suggest renaming this to `Canceller`.

Reply:
done